system/jlib/jutil.cpp

```diff
@@ -3030,19 +3051,117 @@ int getEnum(const char *v, const EnumMapping *map, int defval)
+{
+    if (isEmptyString(authority)||authorityLen==0)
+        return;
+    const char *at = (const char *) memchr(authority, '@', authorityLen);
+    if (!at)
+        splitUrlAddress(authority, authorityLen, host, port);
+    else
+    {
+        size_t userinfoLen = (at - authority);
+        splitUrlAddress(at+1, authorityLen - userinfoLen - 1, host, port);
+        const char *sep = (const char *) memchr(authority, ':', at - authority);
+        if (!sep)
+            user.append(at-authority, authority);
+        else
+        {
+            user.append(sep-authority, authority);
+            size_t passwordLen = (at - sep - 1);
+            password.append(passwordLen, sep+1);
+        }
+    }
+}
+
+static inline void extractUrlProtocol(const char *&url, bool &https, StringBuffer *scheme)
+{
+    if(!url || strlen(url) <= 7)
+        throw MakeStringException(-1, "Invalid URL %s", url);
+    if (0 == strncasecmp(url, "HTTPS://", 8))
+        https = true;
+    else if (0 == strncasecmp(url, "HTTP://", 7))
+        https = false;
+    else
+        throw MakeStringException(-1, "Invalid URL, no protocol specified %s", url);
+    if (https)
+    {
+        url+=8;
+        if (scheme)
+            scheme->append("https://");
+    }
+    else
+    {
+        url+=7;
+        if (scheme)
+            scheme->append("http://");
+    }
+}
+
+static void splitUrlSections(const char *url, bool &https, const char * &authority, size_t &authorityLen, StringBuffer &fullpath, StringBuffer *scheme)
+{
+    extractUrlProtocol(url, https, scheme);
+    const char* path = strchr(url, '/');
+    authority = url;
+    if (!path)
+        authorityLen = strlen(authority);
+    else
+    {
+        authorityLen = path-url;
+        fullpath.append(path);
+    }
+}
+
+extern jlib_decl void splitFullUrl(const char *url, bool &https, StringBuffer &user, StringBuffer &password, StringBuffer &host, StringBuffer &port, StringBuffer &path)
+{
+    const char *authority = nullptr;
+    size_t authorityLen = 0;
+    splitUrlSections(url, https, authority, authorityLen, path, nullptr);
+    splitUrlAuthority(authority, authorityLen, user, password, host, &port);
+}
+
+extern jlib_decl void splitUrlSchemeHostPort(const char *url, StringBuffer &user, StringBuffer &password, StringBuffer &schemeHostPort, StringBuffer &path)
+{
+    bool https;
+    const char *authority = nullptr;
+    size_t authorityLen = 0;
+    splitUrlSections(url, https, authority, authorityLen, path, &schemeHostPort);
+    splitUrlAuthority(authority, authorityLen, user, password, schemeHostPort, nullptr);
+}
+
+//---------------------------------------------------------------------------------------------------------------------
+
+
 static StringBuffer secretDirectory;
 static CriticalSection secretCS;
-static unsigned secretTimeoutMs = UINT_MAX;
+static unsigned secretTimeoutMs = 60 * 60 * 1000;
```

Comment:
Is this typical for k8s?  (I never researched what the standard default was.)

Reply:
I'll see if there's some consensus somewhere.